src/server/graphql/types/InvoiceLineItem.js

```diff
@@ -0,0 +1,42 @@
+import {
+  GraphQLFloat,
+  GraphQLID,
+  GraphQLInt,
+  GraphQLList,
+  GraphQLNonNull,
+  GraphQLObjectType,
+  GraphQLString
+} from 'graphql';
+import InvoiceLineItemEnum from 'server/graphql/types/InvoiceLineItemEnum';
+import InvoiceLineItemDetails from 'server/graphql/types/InvoiceLineItemDetails';
+
+
+const InvoiceLineItem = new GraphQLObjectType({
+  name: 'InvoiceLineItem',
+  description: 'A single line item charge on the invoice',
+  fields: () => ({
+    id: {type: new GraphQLNonNull(GraphQLID), description: 'The unique line item id'},
+    amount: {
+      type: new GraphQLNonNull(GraphQLFloat),
+      description: 'The amount for the line item (in USD)'
+    },
+    description: {
+      type: GraphQLString,
+      description: 'A description of the charge. Only present if we have no idea what the charge is'
```

Comment:
Ah! This is because stripe invoices are ugly as sin. For example, if you add a new team member, they will say 
```
subscription x 5, prorated.... $3.21
subscription x 5, refunded.... $1.24
subscription x 6, prorated... $2.56
```

so uhh, who joined the team? how do i know if someone joined or just unpaused?
what i do is map that to our knowledge of who joined/left/paused/unpaused & create a clean invoice. However, maybe we apply a coupon, or stripe applies some modification that i've never heard of & didn't account for. I still want to show that line so the total adds up to what they were charged, but that line item will be an ugly line coming straight from stripe.

Reply:
Ok, I think I see what you're getting at. Though, peeking at the [stripe invoices docs](https://stripe.com/docs/subscriptions/invoices) (for the first time, heh), aren't _we_ the ones generating line items in the first place? If we post new line items to stripe, don't we get to provide a description at the time of creation, and then just reference that here?

Reply:
nope, stripe is really lacking in this department.
so, you click pause, the backend generates a `prorationDate`. then, it tells stripe to remove that user from the org at the given prorationDate. Stripe does that, which generates an invoiceitem, which it then tells us about via invoiceitem.created webhook. We look up that invoiceitem using the prorationDate as a pseudo foreign key (this is the ONLY way to do it. i talked to the stripe team forever about this. it's wonky.) and when we match up their invoiceitem with the one that we have, i attach the metadata to it so even if our database goes under, we have that data in stripe. Once the metadata is attached, we know what that invoiceitem refers to.

Reply:
K, thanks for the thorough explanation. Sounds like this is the best option we've got!